### PR TITLE
Support setting VF mac address

### PR DESCRIPTION
**Pull request by steevenlee**
This set mac feature works only work when VF not started, in other words, to setup VF mac, has to stop testpmd that binds to VF. Configure on-the-fly wouldn't work, is this the expected model?

### CODE REVIEW

**Comment by ajitkhaparde** [comment]
I think so. 
Probably that is because generally initialization is the only way for hardware/devices
to register a MAC address with the applications/stack.

For dynamic MAC change, the request needs to come from the VF.

File: drivers/net/bnxt/bnxt_cpr.c [hunk 1]
```diff
@@ -87,7 +87,7 @@ void bnxt_handle_fwd_req(struct bnxt *bp, struct cmpl_base *cmpl)
 		req_len = sizeof(fwreq->encap_request);
 
 	/* Locate VF's forwarded command */
-	fwd_cmd = (struct input *)bp->pf.vf_info[vf_id - bp->pf.first_vf_id].req_buf;
+	fwd_cmd = (struct input *)bp->pf.vf_info[fw_vf_id - bp->pf.first_vf_id].req_buf;
```
Comment (stephen-hurd) [inline drivers/net/bnxt/bnxt_cpr.c:90]: Fixes like these should be a separate pull request.

The fix has been included in a separate merge, so a rebase of this branch should resolve this one.

File: drivers/net/bnxt/rte_pmd_bnxt.c [hunk 2]
```diff
@@ -62,7 +62,8 @@ int rte_pmd_bnxt_set_tx_loopback(uint8_t port, uint8_t on)
 	bp = (struct bnxt *)eth_dev->data->dev_private;
 
 	if (!BNXT_PF(bp)) {
-		RTE_LOG(ERR, PMD, "Attempt to operate on none-PF!\n");
+		RTE_LOG(ERR, PMD, "Attempt to set PF loopback on none-PF port %d!\n",
```
Comment (stephen-hurd) [inline drivers/net/bnxt/rte_pmd_bnxt.c]: Should be either "non-PF" or "VF".

File: drivers/net/bnxt/rte_pmd_bnxt.c [hunk 3]
```diff
@@ -121,6 +122,35 @@ int rte_pmd_bnxt_set_all_queues_drop_en(uint8_t port, uint8_t on)
...
+
+	dev = &rte_eth_devices[port];
+	rte_eth_dev_info_get(port, &dev_info);
+	bp = (struct bnxt *)dev->data->dev_private;
+
+	if (vf >= dev_info.max_vfs || mac_addr == NULL)
+		return -EINVAL;
+
+	if (!BNXT_PF(bp)) {
+		RTE_LOG(ERR, PMD, "Attempt to set VF %d mac address on none-PF port %d!\n",
```
Comment (stephen-hurd) [inline drivers/net/bnxt/rte_pmd_bnxt.c]: As above... non-PF or VF

**Review by stephen-hurd — CHANGES_REQUESTED** [review]
Minor changes required.  Looks good otherwise.

**Review by stephen-hurd — APPROVED** [review]

**Pull request merged**